packages/server/src/fhir/repo.ts

```diff
@@ -509,6 +509,12 @@ export class Repository extends BaseRepository implements FhirRepository<PoolCli
     if (!validator.isUUID(id)) {
       throw new OperationOutcomeError(badRequest('Invalid id'));
     }
+    if (resource.resourceType !== resourceType) {
+      throw new OperationOutcomeError(badRequest('Incorrect resource type'));
+    }
+    if (resource.id !== id) {
+      throw new OperationOutcomeError(badRequest('Incorrect ID'));
+    }
```

Comment:
These checks can never fail:
```ts
const { resourceType, id } = resource; // <-- resourceType and id come from resource
// ...
if (resource.resourceType !== resourceType) { // <-- comparing back to resource must succeed
  throw new OperationOutcomeError(badRequest('Incorrect resource type'));
}
if (resource.id !== id) { // <-- same here
  throw new OperationOutcomeError(badRequest('Incorrect ID'));
}
```

In order to make this work correctly, the updated `resourceType` and `id` should be compared against the existing ones in a stored resource to ensure they will not be changed in the update.  This might require some deeper refactoring; I can take a look in the next day or two to evaluate

Reply:
Ah ok, I was trying to add some specs for that change and running into that problem - I'll start poking at it on my end